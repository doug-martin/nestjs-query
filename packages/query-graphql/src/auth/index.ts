export * from './authorizer';
export * from './default-crud.authorizer';
export * from './tokens';
