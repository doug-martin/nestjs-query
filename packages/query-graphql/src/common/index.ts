export { DTONamesOpts, getDTONames, DTONames, getDTOIdTypeOrDefault } from './dto.utils';
export * from './external.utils';
export * from './resolver.utils';
export * from './object.utils';
