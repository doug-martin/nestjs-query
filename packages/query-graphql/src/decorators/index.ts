export {
  FilterableField,
  FilterableFieldOptions,
  FilterableFieldDescriptor,
  getFilterableFields,
} from './filterable-field.decorator';
export { ResolverMethodOpts } from './resolver-method.decorator';
export {
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  Relation,
  FilterableRelation,
  RelationDecoratorOpts,
  RelationTypeFunc,
  getRelations,
} from './relation.decorator';
export * from './resolver-mutation.decorator';
export * from './resolver-query.decorator';
export * from './resolver-field.decorator';
export { Reference, ReferenceDecoratorOpts, ReferenceTypeFunc } from './reference.decorator';
export { ResolverSubscription, SubscriptionResolverMethodOpts } from './resolver-subscription.decorator';
export { InjectPubSub } from './inject-pub-sub.decorator';
export * from './skip-if.decorator';
export * from './aggregate-query-param.decorator';
export * from './hook.decorator';
export * from './decorator.utils';
export * from './hook-args.decorator';
export * from './authorizer.decorator';
export * from './inject-authorizer.decorator';
export * from './inject-custom-authorizer.decorator';
export * from './key-set.decorator';
export * from './authorize-filter.decorator';
export * from './query-options.decorator';
export * from './id-field.decorator';
