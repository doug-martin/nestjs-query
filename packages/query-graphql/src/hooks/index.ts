export * from './hooks';
export { createDefaultHook } from './default.hook';
export * from './tokens';
export * from './types';
