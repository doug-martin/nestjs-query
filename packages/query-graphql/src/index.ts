export * from './types';
export {
  FilterableField,
  FilterableFieldOptions,
  ResolverMethodOpts,
  Relation,
  FilterableRelation,
  CursorConnection,
  FilterableCursorConnection,
  OffsetConnection,
  FilterableOffsetConnection,
  UnPagedRelation,
  FilterableUnPagedRelation,
  RelationTypeFunc,
  RelationDecoratorOpts,
  Reference,
  ReferenceTypeFunc,
  ReferenceDecoratorOpts,
  InjectPubSub,
  BeforeCreateOne,
  BeforeCreateMany,
  BeforeUpdateOne,
  BeforeUpdateMany,
  BeforeDeleteOne,
  BeforeDeleteMany,
  BeforeQueryMany,
  BeforeFindOne,
  InjectAuthorizer,
  InjectCustomAuthorizer,
  Authorize,
  AuthorizerFilter,
  RelationAuthorizerFilter,
  HookArgs,
  MutationHookArgs,
  KeySet,
  QueryOptions,
  IDField,
  IDFieldOptions,
} from './decorators';
export * from './resolvers';
export * from './federation';
export { DTONamesOpts } from './common';
export { NestjsQueryGraphQLModule } from './module';
export { AutoResolverOpts } from './providers';
export { pubSubToken, GraphQLPubSub } from './subscription';
export { Authorizer, CustomAuthorizer, AuthorizerOptions, AuthorizationContext, OperationGroup } from './auth';
export {
  Hook,
  HookTypes,
  BeforeCreateOneHook,
  BeforeCreateManyHook,
  BeforeUpdateOneHook,
  BeforeUpdateManyHook,
  BeforeDeleteOneHook,
  BeforeDeleteManyHook,
  BeforeQueryManyHook,
  BeforeFindOneHook,
} from './hooks';
export { AuthorizerInterceptor, AuthorizerContext, HookInterceptor, HookContext } from './interceptors';
