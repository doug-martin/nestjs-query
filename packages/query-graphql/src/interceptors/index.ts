export * from './hook.interceptor';
export * from './authorizer.interceptor';
