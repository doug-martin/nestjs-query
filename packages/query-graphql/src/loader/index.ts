export * from './find-relations.loader';
export * from './query-relations.loader';
export * from './count-relations.loader';
export * from './aggregate-relations.loader';
export * from './dataloader.factory';
