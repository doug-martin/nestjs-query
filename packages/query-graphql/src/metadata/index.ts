import { GraphQLQueryMetadataStorage } from './metadata-storage';

export { FilterableFieldDescriptor } from './metadata-storage';

/** @internal */
let storage: GraphQLQueryMetadataStorage;
/** @internal */
export function getMetadataStorage(): GraphQLQueryMetadataStorage {
  if (!storage) {
    storage = new GraphQLQueryMetadataStorage();
  }
  return storage;
}
