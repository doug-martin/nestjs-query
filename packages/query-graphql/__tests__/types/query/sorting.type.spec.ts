// eslint-disable-next-line max-classes-per-file
import { ObjectType, InputType, Query, Resolver, Args, Int } from '@nestjs/graphql';
import { FilterableField } from '../../../src';
import { getOrCreateSortType } from '../../../src/types/query/sorting.type';
import { generateSchema } from '../../__fixtures__';

describe('SortingType', (): void => {
  @ObjectType({ isAbstract: true })
  class BaseType {
    @FilterableField()
    id!: number;
  }

  @ObjectType()
  class TestSort extends BaseType {
    @FilterableField()
    stringField!: string;

    @FilterableField()
    numberField!: number;

    @FilterableField()
    boolField!: boolean;
  }

  it('should create the correct graphql schema for sorting type', async () => {
    @InputType()
    class Sorting extends getOrCreateSortType(TestSort) {}

    @Resolver()
    class SortingTypeSpec {
      @Query(() => Int)
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      test(@Args('input') input: Sorting): number {
        return 1;
      }
    }
    const schema = await generateSchema([SortingTypeSpec]);
    expect(schema).toMatchSnapshot();
  });

  it('should throw an error if the class is not annotated with @ObjectType', () => {
    class BadTestSort {}
    expect(() => getOrCreateSortType(BadTestSort)).toThrow(
      'Unable to make SortType. Ensure BadTestSort is annotated with @nestjs/graphql @ObjectType',
    );
  });
  it('should throw an error if no fields are found', () => {
    @ObjectType()
    class BadTestSort {}
    expect(() => getOrCreateSortType(BadTestSort)).toThrow(
      'No fields found to create SortType for BadTestSort. Ensure fields are annotated with @FilterableField',
    );
  });
});
