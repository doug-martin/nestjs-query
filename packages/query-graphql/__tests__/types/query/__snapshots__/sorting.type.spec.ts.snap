// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingType should create the correct graphql schema for sorting type 1`] = `
type Query {
  test(input: Sorting!): Int!
}

input Sorting {
  field: TestSortSortFields!
  direction: SortDirection!
  nulls: SortNulls
}

enum TestSortSortFields {
  id
  stringField
  numberField
  boolField
}

"""Sort Directions"""
enum SortDirection {
  ASC
  DESC
}

"""Sort Nulls Options"""
enum SortNulls {
  NULLS_FIRST
  NULLS_LAST
}

`;
